pkg/operator/ceph/spec/keyring.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2018 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package spec provides Kubernetes controller/pod/container spec items used for many Ceph daemons
+package spec
+
+import (
+	"fmt"
+
+	"github.com/rook/rook/pkg/clusterd"
+	"github.com/rook/rook/pkg/daemon/ceph/client"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	KeyringSecretKeyName = "keyring"
+)
+
+type KeyringConfig struct {
+	Namespace    string
+	ResourceName string
+	DaemonName   string
+	OwnerRef     metav1.OwnerReference
+	Username     string
+	Access       []string
+}
+
+func CreateKeyring(context *clusterd.Context, config KeyringConfig) error {
```

Comment:
Is there also a `DeleteKeyring` function? :slightly_smiling_face:

Reply:
One is needed